Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -1571,50 +1603,59 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
 
 RefPtr<WebExtensionWindow> WebExtensionContext::frontmostWindow(IgnoreExtensionAccess ignoreExtensionAccess) const
 {
-    if (!m_openWindowIdentifiers.isEmpty())
-        return getWindow(m_openWindowIdentifiers.first(), std::nullopt, ignoreExtensionAccess);
+    if (!m_windowOrderVector.isEmpty())
+        return getWindow(m_windowOrderVector.first(), std::nullopt, ignoreExtensionAccess);
     return nullptr;
 }
 
-void WebExtensionContext::didOpenWindow(const WebExtensionWindow& window)
+void WebExtensionContext::didOpenWindow(WebExtensionWindow& window, UpdateWindowOrder updateWindowOrder, SuppressEvents suppressEvents)
 {
-    ASSERT(window.extensionContext() == this);
-    ASSERT(m_windowMap.contains(window.identifier()));
-    ASSERT(m_windowMap.get(window.identifier()) == &window);
+    ASSERT(isValidWindow(window));
+
+    // The window might already be open, don't log an error.
```

Comment:
We blindly call this in a couple places, to make sure it is open, like the delegate calls to populate on load, when the app might have already told us about the window directly too. These methods are also tied directly to the API, and I figured it was fine to no show too much noise if client app calls it multiple times since we ignore repeats.